### DISCUSSION TITLE
Parse schema without validate

### DISCUSSION THREAD

**Discussion by victj99** [colinhacks/zod#2438] [Q&A] [answered]
Hello, i'm new in zod and is a amazing library, i'm testing some cases and i have not found a way to parse a schema without validate like yup cast.
Example

``` typescript
import * as yup from 'yup';
import { z } from 'zod'

const yupSchema = yup.object({
	rowsPerPage: yup.number().min(5),
	page: yup.number().min(10),
	sortBy: yup.string().email().optional(),
})

const zodSchema = z.object({
	rowsPerPage: z.coerce.number().min(5),
	page: z.coerce.number().min(10),
	sortBy: z.string().email().optional(), // email only for test
})

const zodParsed = zodSchema.parse({ rowsPerPage: "2", page: 1, sortBy: 'id'}) // error by min and email
const yupCast =  yupSchema.cast({ rowsPerPage: "2", page: 1, sortBy: 'id'}) // returns { rowsPerPage: 2, page: 1, sortBy: 'id'}

// cast and validate
const yupValidate =  yupSchema.validate({ rowsPerPage: "2", page: 1, sortBy: 'id'}) // error by min and email
```

I only want parse a json in some cases and validate it later.
It is possible?
thanks

**Comment by mrtnprzk** [comment]
Hello. This is exactly what I am looking for right now. Any update if is possible to do that and how?

**Comment by JacobWeisenburger** [comment]
Is this what you are looking for?

```ts
const testData = { rowsPerPage: '2', page: 1, sortBy: 'id' }

const yupSchema = yup.object( {
    rowsPerPage: yup.number().min( 5 ),
    page: yup.number().min( 10 ),
    sortBy: yup.string().email().optional(),
} )

const yupCast = yupSchema.cast( testData )
console.log( yupCast )
// {
//     sortBy: "id",
//     page: 1,
//     rowsPerPage: 2
// }

const looseZodSchema = z.object( {
    rowsPerPage: z.coerce.number(),
    page: z.coerce.number(),
    sortBy: z.coerce.string(),
} )

const looseData = looseZodSchema.parse( testData )
console.log( looseData )
// {
//     rowsPerPage: 2,
//     page: 1,
//     sortBy: "id"
// }
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger

**Comment by mrtnprzk** [comment]
I do not think so. We use zod for validating inputs in react app also with react-hook-form.
Atm. we have two objects:

```ts
const defaultValue = {
    description: '',
};

const schema = {
    description: z.string().min(1),
};
```

Because these object are really huge I had idea to use .default() to define defaultValues and parse it to get object with default values.
Something like this:

```ts
const schema = {
    description: z.string().min(1).default(''),
};

const defaultValue = schema.parse({});
```

However I am getting error because empty string will not pass min(1).
So I was wondering if there is no possibility how to do that.

**Reply by JacobWeisenburger** [reply, marked as answer]
as far as I know, zod doesn't have anything that allows you to use the same schema for "casting" and also for "validating". You would need to make a schema for "casting" and a schema for "validating".
